src/hybrid.jl

```diff
@@ -3,67 +3,77 @@
+"""
+function train{C, T<: AbstractFloat, U <: Integer, N}(::Type{HybridNB}, continuous_features::Dict{N, Vector{T}}, discrete_features::Dict{N, Vector{U}}, labels::Vector{C})
+    return fit(HybridNB(labels, N), continuous_features, discrete_features, labels)
+end
+
 
 """
     fit(m::HybridNB, f_c::Matrix{Float64}, labels::Vector{Int64})
 
 Train NB model with continuous features only
 """
 function fit{C, T<: AbstractFloat}(model::HybridNB, continuous_features::Matrix{T}, labels::Vector{C})
-    discrete_features = Vector{Vector{Int64}}()
+    discrete_features = Dict{Symbol, Vector{Int64}}()
     return fit(model, restructure_matrix(continuous_features), discrete_features, labels)
 end
 
 
 """computes log[P(x⃗ⁿ|c)] ≈ ∑ᵢ log[p(xⁿᵢ|c)] """
-function sum_log_x_given_c!{T <: AbstractFloat, U <: Integer}(class_prob::Vector{Float64}, feature_prob::Vector{Float64}, m::HybridNB, continuous_features::Vector{Vector{T}}, discrete_features::Vector{Vector{U}}, c)
+function sum_log_x_given_c!{T <: AbstractFloat, U <: Integer, N}(class_prob::Vector{Float64}, feature_prob::Vector{Float64}, m::HybridNB, continuous_features::Dict{N, Vector{T}}, discrete_features::Dict{N, Vector{U}}, c)
     for i = 1:num_samples(m, continuous_features, discrete_features)
-
-        for j = 1:num_kdes(m)
-            if isnan(continuous_features[j][i])
+        for (j, name) in enumerate(keys(continuous_features))
+            x_i = continuous_features[name][i]
+            if isnan(x_i)
                 feature_prob[j] = NaN
             else
-                feature_prob[j] = pdf(m.c_kdes[c][j], continuous_features[j][i])
+                feature_prob[j] = pdf(m.c_kdes[c][name], x_i)
             end
         end
 
-        for j = 1:num_discrete(m)
-            if isnan(continuous_features[j][i])
+        for (j, name) in enumerate(keys(discrete_features))
+            x_i = discrete_features[name][i]
+            if isnan(x_i)
                 feature_prob[num_kdes(m)+j] = NaN
             else
-                feature_prob[num_kdes(m)+j] = probability(m.c_discrete[c][j], discrete_features[j][i])
+                feature_prob[num_kdes(m)+j] = probability(m.c_discrete[c][name], x_i)
             end
         end
-
         sel = isfinite(feature_prob)
         class_prob[i] = sum(log(feature_prob[sel]))
     end
 end
 
 
 """ compute the number of samples """
-function num_samples{T <: AbstractFloat, U <: Integer}(m::HybridNB, continuous_features::Vector{Vector{T}}, discrete_features::Vector{Vector{U}}) # TODO: this is a bit strange
-    if num_kdes(m) > num_discrete(m)
-        n_samples = length(continuous_features[1])
+function num_samples{T <: AbstractFloat, U <: Integer, N}(m::HybridNB, continuous_features::Dict{N, Vector{T}}, discrete_features::Dict{N, Vector{U}}) # TODO: this is a bit strange
+    if length(keys(continuous_features)) > 0
+        return length(continuous_features[collect(keys(continuous_features))[1]])
+    elseif length(keys(discrete_features)) > 0
```

Comment:
no need for `elseif`, just do:

    if length(keys(continuous_features)) > 0
        return length(continuous_features[collect(keys(continuous_features))[1]])
    end

    if length(keys(discrete_features)) > 0
        return length(discrete_features[collect(keys(discrete_features))[1]])
    end

    return 0